eva_cttv_pipeline/clinvar.py

```diff
@@ -227,57 +101,96 @@ def observed_refs_list(self):
                 for ref in self.observed_pubmed_refs]
 
     @property
-    def measure_set_refs_list(self):
-        return ['http://europepmc.org/abstract/MED/' + str(ref)
-                for ref in self.measure_set_pubmed_refs]
+    def clinical_significance(self):
+        return \
+            self.data['referenceClinVarAssertion']['clinicalSignificance']['description'].lower()
+
+    @property
+    def allele_origins(self):
+        allele_origins = set()
+        for clinvar_assetion_document in self.data['clinVarAssertion']:
+            for observed_in_document in clinvar_assetion_document['observedIn']:
+                allele_origins.add(observed_in_document['sample']['origin'].lower())
+
+        return list(allele_origins)
+
+
+class ClinvarRecordMeasure(UserDict):
+
+    def __init__(self, clinvar_measure_dict, clinvar_record):
+        UserDict.__init__(self, clinvar_measure_dict)
+        self.clinvar_record = clinvar_record
+
+    @property
+    def rs_id(self):
+        if "xref" in self.data:
+            for xref in self.data["xref"]:
+                if xref["db"].lower() == "dbsnp":
+                    return "rs{}".format(xref["id"])
+        return None
+
+    @property
+    def nsv_id(self):
+        if "xref" in self.data:
+            for xref in self.data["xref"]:
+                if xref["db"].lower() == "dbvar":
+                    return xref["id"]
+        return None
 
     @property
     def hgvs(self):
         hgvs_list = []
-        for measure in self.data['referenceClinVarAssertion']['measureSet']['measure']:
-            for attribute_set in measure['attributeSet']:
-                if attribute_set['attribute']['type'].startswith('HGVS'):
-                    hgvs_list.append(attribute_set['attribute']['value'])
+        for attribute_set in self.data['attributeSet']:
+            if attribute_set['attribute']['type'].startswith('HGVS'):
+                hgvs_list.append(attribute_set['attribute']['value'])
 
         return hgvs_list
 
     @property
-    def clinical_significance(self):
-        return \
-            self.data['referenceClinVarAssertion']['clinicalSignificance']['description'].lower()
+    def variant_type(self):
+        return self.data['type']
 
-    def __get_rs(self, rcv_to_rs):
-        try:
-            return rcv_to_rs[self.accession]
-        except KeyError:
-            return None
+    @property
+    def measure_set_pubmed_refs(self):
```

Comment:
I can change it to the even simpler "pubmed_refs" since it's already part of the ClinvarRecordMeasure class.